### ISSUE TITLE
String like 01234 is parsed as number in YAML - regression

### ISSUE THREAD

**Issue by dejvid-smth** [stoplightio/spectral#1345]
**Describe the bug**
When linting YAML OpenAPI 3.x schema string value that looks like number but starts with 0 is incorrectly considered to be a number. This produces linting error even though the document is correct.

**To Reproduce**

1. Given this OpenAPI document 
```yml
openapi: '3.0.2'
info:
  title: API Title
  version: '1.0'
servers:
  - url: https://api.server.test/v1
paths:
  /test:
    get:
      responses:
        '200':
          description: OK
          content:
            'application/json':
              schema:
                type: object
                properties:
                  value:
                    type: string
                    example: 0125487936
```
2. Run this CLI command `spectral lint the_document.yml`
3. I get this error ``20:30    error  oas3-valid-schema-example  `example` property type should be string``

**Expected behavior**
I would expect that the error is not produced and the schema is considered valid. This is how version 5.3.0 behaves.

**Environment:**
 - Library version: 5.5.0

**Comment by nulltoken** [comment]
@dejvid-smth :wave: As noted in the [doc](https://github.com/stoplightio/spectral/blob/4116a58296938c72704a5384c1cb9889afee9f5f/docs/guides/3-javascript.md#linting-a-yaml-string): _'Please note that by default Spectral supports YAML 1.2 with merge keys extension. As recommended by the spec, tags are resolved according to the "Core Schema."'_

According to [Core schema](https://yaml.org/spec/1.2/spec.html#id2804923) , `0125487936` would indeed not be a string but a number. Would you ensure it to be parsed as a string, surrounding it with double quotes should do the trick.

@philsturgeon @P0lip That YAML format expectation may actually be buried a little _too_ deep in the doco. Where could we move it so that users are well informed about it when crafting their YAML rulesets or linting their YAML documents?

**Comment by P0lip** [comment]
@nulltoken hmm, good question. I forgot we actually had such a statement, since it's indeed quite hidden.
I cannot think of any other spot, to be honest. @philsturgeon do you have any clue?

**Comment by P0lip** [comment]
@nulltoken perhaps a separate document about YAML would be a good choice here? We could dump more examples there.

**Comment by nulltoken** [comment]
> @nulltoken perhaps a separate document about YAML would be a good choice here? We could dump more examples there.

As YAML can be used both for rulesets and documents, that would be an amazing idea!
